File: algorithm/504. Base 7/504.py
class Solution(object):
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        x=abs(num)
        res=""
        while True:
            tem=x%7
            res=str(tem)+res
            x/=7
            if x==0:
                break
        if num<0:
            return ("-"+res)
        return res

File: algorithm/504. Base 7/504.cpp
class Solution {
public:
    string convertToBase7(int num) {
        int tem,flag=num;
        num=abs(num);
        string res="";
        while(num>=7)
        {
            tem=num%7;
            res=to_string(tem)+res;
            num/=7;
        }
        tem=num%7;
        res=to_string(tem)+res;
        return flag>=0 ? res:("-"+res);
    }
};
